frontend/src/app/(vendor)/vendors/login/page.tsx

```diff
@@ -0,0 +1,122 @@
+"use client";
+
+import Container from "@mui/material/Container";
+import Typography from "@mui/material/Typography";
+import CircularProgress from "@mui/material/CircularProgress";
+import Box from "@mui/material/Box";
+import { useEffect, useRef, useState } from "react";
+import { useRouter, useSearchParams } from "next/navigation";
+import { createClient } from "@/lib/supabase/client";
+import { VendorLoginForm } from "@/features/login/components/VendorLoginForm";
+import { fetchVendorBySlug } from "@/features/profile/common/api/fetchVendor";
+import { ReCaptchaRef } from '@/components/security/ReCaptcha';
+
+export default function VendorLoginPage() {
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const recaptchaRef = useRef<ReCaptchaRef>(null);
+
+  // Extract magic link parameters
+  const slug = searchParams.get('slug') || "";
+  const email = searchParams.get('email') || "";
+  const token = searchParams.get('token') || "";
+
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (process.env.NEXT_PUBLIC_FEATURE_VENDOR_LOGIN_ENABLED !== 'true') {
+      router.push(`/`);
+      return;
+    }
+
+    // Check if user is already logged in
+    const checkSession = async () => {
+      const supabase = createClient();
+      const { data: { session } } = await supabase.auth.getSession();
+
+      if (session) {
+        console.log("User is already logged in");
+        // User is already logged in, redirect to vendor page
+        router.push(`/`);
+        return;
+      }
+
+      const areAllParamsValid = !!email && !!token && email.trim() !== "" && token.trim() !== "" && slug.trim() !== "";
+
+      if (!areAllParamsValid) {
+        // Email, token, or slug is missing, show the login form
+        setIsLoading(false);
+        return;
+      }
+
+      // Execute reCAPTCHA and get token
+      try {
+        await recaptchaRef.current?.executeAsync();
+        console.log("reCAPTCHA executed successfully");
+      } catch (error) {
+        console.error("Error executing reCAPTCHA: ", error);
+        setIsLoading(false);
+        return;
+      }
+
+      // Check if email and token from the query parameters are valid and match database records. 
+      // If they do, sign in the user anonymously and link their email to the account.
+      const vendor = await fetchVendorBySlug(slug);
+      const doEmailAndTokenMatch = email === vendor?.email?.toLowerCase() && token === vendor?.access_token?.toLowerCase();
+
+      if (doEmailAndTokenMatch) {
+        // Sign in the user anonymously and link email
+        const { data, error } = await supabase.auth.signInAnonymously()
+        if (error) {
+          console.error("Error signing in anonymously: " + error.message);
+          setIsLoading(false);
+          return;
+        }
+
+        const { data: updateEmailData, error: updateEmailError } = await supabase.auth.updateUser({
```

Comment:
this triggers a verification email right? should we notify the user?

Reply:
if they don't verify, they can still proceed with edits and it's not blocking? But the next time they login with email, they'll get the normal login error flow when email isn't verified?

just wondering if we should block the editing experience on verification. If my understanding is correct, then we don't need to block